vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListImagesController.java

```diff
@@ -65,4 +73,45 @@ public String listSpaces(@PathVariable(required = false) String page,
                 (order==null || order.equals("")) ? Sort.Direction.DESC.toString().toLowerCase():order);
         return "staff/images/imagelist";
     }
+    
+    @RequestMapping("/staff/images/list/tag/{filter}/{page}")
+    public String listSpacesByTag(@PathVariable(required = false) String page,
+            @PathVariable(required = false) String filter,
+            @RequestParam(value = "sort", required = false) String sortedBy,
+            @RequestParam(value = "order", required = false) String order, Model model) {
+        int pageNo;
+        page = StringUtils.isEmpty(page) ? "1" : page;
```

Comment:
switch with line 82 (declare variables as close to where they are needed as possible)